pkg/secrethub/iterator/paginator.go

```diff
@@ -0,0 +1,31 @@
+package iterator
+
+type paginator struct {
+	fetched bool
+	fetch   func() ([]interface{}, error)
+}
+
+// PaginatorConstructorWithFetch returns a new paginator constructor that constructs a paginator
+// with the provided fetch function.
+func PaginatorConstructorWithFetch(fetch func() ([]interface{}, error)) PaginatorConstructor {
```

Comment:
Maybe call this `NewPaginatorFactory`? Or `NewSinglePagePaginatorFactory` (as all paginators fetch pages, the unique thing about this paginator is that is fetches everything in one page.

Reply:
Guys, just saw this pass by and it looks like you're creating a ConstructorConstructor to inject fetch functionality. This seems pretty convoluted and looks like it's for testing purposes. 

If that's the case, think carefully about exposing a public function in this package only for testing purposes.